website/docs/r/nsxt_firewall.html.markdown

```diff
@@ -0,0 +1,116 @@
+
+  rule {
+    name        = "allow all IPv4 traffic"
+    direction   = "IN_OUT"
+    ip_protocol = "IPV4"
+  }
+}
+```
+
+## Example Usage 2 (Multiple firewall rules - order matters)
+```hcl
+resource "vcd_nsxt_firewall" "testing" {
+  org  = "my-org"
+  vdc  = "my-nsxt-vdc"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.testing.id
+
+  # Rule #1 - Allows in IPv4 traffic from security group `vcd_nsxt_security_group.group1.id`
+  rule {
+    name        = "first rule"
+    direction   = "IN"
+    ip_protocol = "IPV4"
+    sources     = [vcd_nsxt_security_group.frontend.id]
+  }
+
+  # Rule #2 - Drops and logs all outgoing IPv6 traffic to `vcd_nsxt_security_group.group.2.id`
+  rule {
+    name         = "drop IPv6 with destination to security group 2"
+    direction    = "OUT"
+    ip_protocol  = "IPV6"
+    destinations = [vcd_nsxt_security_group.group2.id]
+    action       = "DROP"
+    logging      = true
+  }
+  
+  # Rule #3 - Allows IPv4 and IPv6 traffic in both directions:
+  # from vcd_nsxt_security_group.group.1.id to all list of security groups vcd_nsxt_security_group.group.*.id
+  # from list of security groups vcd_nsxt_security_group.group.*.id to vcd_nsxt_security_group.group.1.id
+  rule {
+    name         = "test_rule-3"
+    direction    = "IN_OUT"
+    ip_protocol  = "IPV4_IPV6"
+    sources      = [vcd_nsxt_security_group.group.1.id]
+    destinations = vcd_nsxt_security_group.group.*.id
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional) The name of organization to use, optional if defined at provider level. Useful
+  when connected as sysadmin working across different organisations.
+* `vdc` - (Optional) The name of VDC to use, optional if defined at provider level.
+* `edge_gateway_id` - (Required) The ID of the edge gateway (NSX-T only). Can be looked up using
+  `vcd_nsxt_edgegateway` datasource
+* `rule` (Required) One or more blocks with [Firewall Rule](#firewall-rule) definitions
+
+<a id="firewall-rule"></a>
+## Firwall Rule
+
+Each Firewall Rule contains following attributes:
+
+* `name` - (Required) Explanatory name for firewall rule (uniqueness not enforced)
+* `direction` - (Required) One of `IN`, `OUT`, or `IN_OUT`
+* `ip_protocol` - (Required) One of `IPV4`,  `IPV6`, or `IPV4_IPV6`
+* `enabled` - (Optional) Defines if the rule is enabled (default `true`)
+* `logging` - (Optional) Defines if logging for this rule is enabled (default `false`)
+* `action` - (Optional) Defines if it should `ALLOW` or `DROP` traffic (default `ALLOW`)
+* `sources` - (Optional) A set of source object Firewall Groups (`IP Sets` or `Security groups`). 
+Leaving it empty matches `Any` (all)
+* `destinations` - (Optional) A set of source object Firewall Groups (`IP Sets` or `Security groups`). 
+Leaving it empty matches `Any` (all)
+* `applications` - (Optional) A set of application port profiles. Leaving it empty matches  (all)
```

Comment:
The ending " (all)" seems abrupt.

Reply:
True. I have rephrased it to the same as in `sources` and `destinations` - hopefully this is better.

However - while re-reading this I saw these names again `source`, `destinations` and `applications`. What if I rename them to `source_ids`, `destination_ids` and `app_port_profile_ids`? In fact these are sets of IDs so probably this name makes more sense.

CC @dataclouder

Reply:
I like this suggestion! And we're using that approach in some places, which makes it explicitly clear:

<img width="1237" alt="Screenshot 2021-06-10 at 09 28 47" src="https://user-images.githubusercontent.com/11330689/121475607-4e5b8a00-c9ce-11eb-8bfc-1b30ce31dd05.png">